Source/WebCore/editing/cocoa/EditorCocoa.mm

```diff
@@ -342,7 +342,7 @@ static void maybeCopyNodeAttributesToFragment(const Node& node, DocumentFragment
 
 void Editor::replaceNodeFromPasteboard(Node& node, const String& pasteboardName, EditAction action)
 {
-    if (node.document() != m_document)
+    if (&node.document() != &m_document)
```

Comment:
Does this fix a bug? What did the code end up doing before?

Reply:
Ah yes, I was a bit surprised the previous code was building. I am not sure what it ended up doing. The `CSSParserContext` constructor that takes in a `Document&` is not marked as explicit and I modified CSSParserContext's operator==() in this patch so maybe it was using that?

I can double check.

Reply:
Yes, I was right. I was using CSSParserContext's operator==() :/
I will write a patch to make CSSParserContext's constructors explicit.

Reply:
Amazing! That is so nuts. That may even have been costly enough that we get a tiny performance boost.

Reply:
will land https://github.com/WebKit/WebKit/pull/17356/ first to avoid behavior changes in this large refactoring patch.